torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -313,6 +313,16 @@ class TORCH_API ProcessGroupNCCL : public Backend {
     friend class ProcessGroupNCCL;
   };
 
+  class DebugInfoCallbackStorer {
+   public:
+    DebugInfoCallbackStorer(std::string fileName);
+    virtual ~DebugInfoCallbackStorer();
+    virtual void storeTraceInfoStorer(int rank, const std::string& ncclTrace);
```

Comment:
3 questions about this API

1) should rank be an arg? (why?) (and if so why not put it into ctor)
2) should ncclTrace be an std::string or do we want to make it another type?  (bytes, or, actual structure?)
3) nit: rename?  just simply "write()"?

Reply:
We want the file comes with rank in its name? I mean for examples, if 8 ranks on a host to write to one single file, we will have race condition and the doc will be in a uncertain state (we don't know which rank will write first, right?)

ncclTrace is actually a vector of <uint_8>, I think @zdevito use string just let users to treat them as a string blob, so that users will do `pickle.loads` in python to convert it back to a list of map I believe. Sure, let's use the name of `write()`

Reply:
We should change that api to actually return std::vector<uint8_t> so that it is less confusing. I know it really only is being used in our code, but it's come up a number of times. I just did `std::string` because it was quick.

Reply:
Sure, I will send a separate PR to change it.